### DISCUSSION TITLE
Desktop Platforms: How does one control the main window?

### DISCUSSION THREAD

**Discussion by brunoga** [cogentcore/core#1140] [General]
More specifically, how can one control its size and position on startup? Is there a way to make a window be fullscreen? I saw some mentions about this in the documentation but I could not really find a way to do that (it does not look like it would be through Body).

**Comment by kkoreilly** [comment]
If you have a use case you can share in which you need to control these properties more directly, we would be open to supporting that. Otherwise, this is the current level of support:

The size and position of a window is automatically saved once a user adjusts it, so it will be restored with the same size and position on future startups. For the default size and position for the first startup, it is not recommended to set manually since it is not very cross-platform. However, you can specify a `Min` size on the `Body` using a `Styler` to increase the default window size:

```go
b.Styler(func(s *styles.Style) {
  s.Min.Set(units.Em(500))
})
```

There is also no way to make a window fullscreen by default, since we believe the user should have control over that. If a user makes a window fullscreen, it will remember that for future startups. Also note that we are planning to improve multi-monitor control soon (#997).

**Reply by brunoga** [reply]
I have this code that controls a gimbal that has an associated video feed coming from it. Because of its nature, fullscreen would be the natural way to show this.

Other than that, I understand you are not targeting games (at least for now), but this would be another case where fullscreen is the natural wya to display things.

**Reply by kkoreilly** [reply]
Thank you for clarifying your use case. I will add support for defaulting to fullscreen as part of #997, since it seems like it will be beneficial for various use cases.

**Reply by kkoreilly** [reply]
@brunoga We implemented full support for customizing the monitor, fullscreen, and geometry properties of windows both when starting them and while they are running in #1341 and #1364. See https://github.com/cogentcore/core/issues/997#issuecomment-2530531438 for more information.

**Reply by brunoga** [reply]
@kkoreilly that is great. Will take a look at it soon. Thank you!

**Comment by c1ngular** [comment]
i would love to have full screen window option in multiple monitor settings, for instance, play a video or display chart full screen on secondary monitor.

**Reply by kkoreilly** [reply]
@c1ngular We implemented full support for customizing the monitor, fullscreen, and geometry properties of windows both when starting them and while they are running in #1341 and #1364. See https://github.com/cogentcore/core/issues/997#issuecomment-2530531438 for more information.

**Reply by c1ngular** [reply]
great to hear, thanks

**Comment by heejit** [comment]
One of my application running as Kisok which provide some information it need to full screen

**Reply by kkoreilly** [reply]
What operating system is it on? We are planning to add support for programmatic fullscreen as part of #997 as mentioned above.

**Reply by heejit** [reply]
windows 7 and ubuntu (16.04, 20.04)

**Reply by kkoreilly** [reply]
Yes, it will work once we implement support for programmatic fullscreen as part of #997.

**Reply by kkoreilly** [reply]
@heejit We implemented full support for customizing the monitor, fullscreen, and geometry properties of windows both when starting them and while they are running in #1341 and #1364. See https://github.com/cogentcore/core/issues/997#issuecomment-2530531438 for more information.
